bridges/snowbridge/pallets/ethereum-client/src/lib.rs

```diff
@@ -398,6 +402,15 @@ pub mod pallet {
 				Error::<T>::IrrelevantUpdate
 			);
 
+			// Verify the finalized header gap between the current finalized header and new imported
+			// header is not larger than the sync committee period, otherwise we cannot do
+			// ancestry proofs for execution headers in the gap.
+			ensure!(
+				latest_finalized_state.slot + config::SLOTS_PER_HISTORICAL_ROOT as u64 >=
```

Comment:
Needs saturating arithmetic to protect against overflow.

also the conversion of SLOTS_PER_HISTORICAL_ROOT to u64 can be checked at compile time by using `into()` instead of `as`

Reply:
How can use `.into` at compile time?

```
let slots_per_historical_root: u64 = config::SLOTS_PER_HISTORICAL_ROOT.into();
```

does not work, it wants me to do `try_into()`, which I think is a run time check?

Reply:
Ahh ok, never mind, just us `as` then, which we know will always work since with the wasm target, usize is 64 bits

Reply:
reverted 😄

Reply:
Still needs this change:

```suggestion
				latest_finalized_state.slot.saturating_add(config::SLOTS_PER_HISTORICAL_ROOT as u64) >=
```

Reply:
Apologies. 😅 Applied.